crates/sel4/sys/src/fault/mod.rs

```diff
@@ -21,30 +23,42 @@ impl seL4_Fault {
         length: seL4_Word,
         f: impl Fn(core::ffi::c_ulong) -> seL4_Word,
     ) -> Self {
-        match label {
-            seL4_Fault_tag::seL4_Fault_NullFault => {
-                // TODO
-                // assert!(length == seL4_NullFault_Msg::seL4_NullFault_Length);
-                seL4_Fault_NullFault_Unpacked {}.unsplay()
-            }
-            seL4_Fault_tag::seL4_Fault_CapFault => {
-                // TODO
-                // assert!(length == seL4_CapFault_Msg::seL4_CapFault_Length);
-                seL4_Fault_CapFault_Unpacked {
-                    IP: f(seL4_CapFault_Msg::seL4_CapFault_IP),
-                    Addr: f(seL4_CapFault_Msg::seL4_CapFault_Addr),
-                    InRecvPhase: f(seL4_CapFault_Msg::seL4_CapFault_InRecvPhase),
-                    LookupFailureType: f(seL4_CapFault_Msg::seL4_CapFault_LookupFailureType),
-                    MR4: f(seL4_CapFault_Msg::seL4_CapFault_BitsLeft),
-                    MR5: f(seL4_CapFault_Msg::seL4_CapFault_GuardMismatch_GuardFound),
-                    MR6: f(seL4_CapFault_Msg::seL4_CapFault_GuardMismatch_BitsFound),
+        sel4_cfg_wrap_match! {
+            match label {
+                seL4_Fault_tag::seL4_Fault_NullFault => {
+                    // TODO
+                    // assert!(length == seL4_NullFault_Msg::seL4_NullFault_Length);
+                    seL4_Fault_NullFault_Unpacked {}.unsplay()
+                }
+                seL4_Fault_tag::seL4_Fault_CapFault => {
+                    // TODO
+                    // assert!(length == seL4_CapFault_Msg::seL4_CapFault_Length);
+                    seL4_Fault_CapFault_Unpacked {
+                        IP: f(seL4_CapFault_Msg::seL4_CapFault_IP),
+                        Addr: f(seL4_CapFault_Msg::seL4_CapFault_Addr),
+                        InRecvPhase: f(seL4_CapFault_Msg::seL4_CapFault_InRecvPhase),
+                        LookupFailureType: f(seL4_CapFault_Msg::seL4_CapFault_LookupFailureType),
+                        MR4: f(seL4_CapFault_Msg::seL4_CapFault_BitsLeft),
+                        MR5: f(seL4_CapFault_Msg::seL4_CapFault_GuardMismatch_GuardFound),
+                        MR6: f(seL4_CapFault_Msg::seL4_CapFault_GuardMismatch_BitsFound),
+                    }
+                    .unsplay()
+                }
+                #[sel4_cfg(KERNEL_MCS)]
+                seL4_Fault_tag::seL4_Fault_Timeout => {
+                    assert!(length == seL4_Timeout_Msg::seL4_Timeout_Length as seL4_Word);
+                    seL4_Fault_Timeout_Unpacked {
+                        data: f(seL4_Timeout_Msg::seL4_Timeout_Data as seL4_Word),
+                        consumed: f(seL4_Timeout_Msg::seL4_Timeout_Consumed as seL4_Word),
+                    }
```

Comment:
About these casts, I've opened https://github.com/seL4/seL4/issues/1468. I'm guessing after that issue is addressed, they will no longer be necessary. For now, would you mind replacing the three instances of `as seL4_Word` with `.into()`, for the sake of consistency with the rest of the codebase?

Reply:
Done